### DISCUSSION TITLE
Third Party Apps / "Native" apps

### DISCUSSION THREAD

**Discussion by cheako** [DustinBrett/daedalOS#37] [Ideas]
Start with a github project hosting the "Hello World" console app.  A console app that asks a few questions and then spawns a window.  A multi-window app.  Personally, I'd be interested in using Rust, [winit or a competing glfw](https://crates.io/crates/winit), DearImGUI.

Can I open an issue for an app store?  I'm thinking apps should be curated, but at the same time there should be a way(checkbox or tab) to look at unacknowledged apps.

**Comment by DustinBrett** [comment]
Thanks! I still don't have very good for native/3rd party app support within daedalOS, but it is planned and at that time I will indeed be writing instructions. Until then it could be made easily enough by copying an example from the apps folder.

**Reply by cheako** [reply]
I took a second look and it's clear as mud.  The files in the filesystem are minified and the apps folder are symbolic, those are some version of Window's Shortcut.lnk files: https://github.com/DustinBrett/daedalOS/blob/main/components/apps/SpaceCadet/useSpaceCadet.ts#L34  Tip: if a TS file is under 500 lines and you have 10 that are basically the same, implement that feature using XML.

**Reply by DustinBrett** [reply]
Could you clarify your point about L34, perhaps with an example. Thanks!

**Reply by cheako** [reply]
The entire file could be implemented as XML for 1.  That line is a link...  to a minified javascript file.  So even though there are example apps, they are compiled and not useful for someone wanting to develop apps.

**Reply by DustinBrett** [reply]
Indeed those files have precompiled minified files. "Native apps" would not likely work in the same way. These minified files hide no secrets, they are just dependencies for the libraries the app uses.

**Comment by ghost** [comment]
Hi @DustinBrett,

Have you thought about doing something similar to **Win11React**?

Win11React, another web desktop, lets developers add applications by defining them in a `store.json` file on GitHub. In that file, you include basic app metadata-like the title, icon, and URL-which is then displayed inside an iframe when the app is launched.

**Docs:** [Win11React - Add App](https://win11react-docs.andrewstech.me/docs/Store/add-app/)

It's pretty simple, and it would also let you sandbox the iframe content using the `sandbox` and `allow` attributes.

PS: You could also support not just `<iframe>` apps, but custom web components too.

**Reply by DustinBrett** [reply]
Yes I hope to one day. It's the top item on my [IDEAS.md](https://github.com/DustinBrett/daedalOS/blob/main/IDEAS.md) doc.

Interesting to bring back this post as it's nearly 4 years old. It's been a busy 4 years and especially the last year for me personally. But I still hope to tackle everything in the list one day, maybe with the help of AI once it gets good enough (still isn't as of Feb 2026).
